main.go

```diff
@@ -542,29 +521,19 @@ func statusAction(ctx *cli.Context) (err error) {
 	var systemStatus SystemStatus
 	var machineReadablePrintFunc func(systemStatus *SystemStatus) error
 
-	uiPreferences := getUserInterfacePreferences(ctx)
-
-	// Only JSON file format is supported ATM
 	format := ctx.String("format")
-	if format != "" {
-		switch format {
-		case "json":
-			uiPreferences.MachineReadable = true
-			machineReadablePrintFunc = printJSONStatus
-			// Disable all colors and animations
-			uiPreferences.IsColorful = false
-		default:
-			err := fmt.Errorf(
-				"unsuported machine-readable format: %s (supported formats: %s)",
-				format, "\"json\"",
-			)
-			return cli.Exit(err, 1)
-		}
+	switch format {
+	case "":
+		break
+	case "json":
+		machineReadablePrintFunc = printJSONStatus
+	default:
+		return fmt.Errorf("unsupported format: %s", format)
```

Comment:
It seems that this code path could not happen, because unsupported formats are catched in `Before` function.

Reply:
My idea was to handle it everywhere, because you never know when you'll forget to update one of those places. But sure, I can flip the logic around.